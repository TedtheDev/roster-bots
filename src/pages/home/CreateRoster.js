import React, { Component } from 'react';
import { connect } from 'react-redux';
import { createRoster } from '../../modules/rosters';
import CleanString from '../../utils/cleanStrings';
import styled from 'styled-components';

const CreateRosterDiv = styled.div`
    margin: 5px;
`;

const ErrorMessage = styled.span`
    color: red;
`;

const PlayerLegendDiv = styled.div`
    border: 1px black solid;
    border-radius: 5px;
`;

export class CreateRoster extends Component {
    constructor(props) {
        super(props);

        this.state = { rosterName: '', errorMessage: '' };
    }

    onChangeRosterName = event => {
        this.setState({rosterName: event.target.value});
    }

    onClickCreateRoster = () => {
        if(this.state.rosterName !== '') {
            this.setState({ errorMessage: ''});
            this.props.dispatch(createRoster(CleanString.toPropertyValue(this.state.rosterName)));
            this.setState({rosterName: '' });
        } else {
            this.setState({ errorMessage: 'Enter a Roster name'});
        }
    }

    componentWillReceiveProps(nextProps) {
        if(nextProps.errorMessage !== this.props.errorMessage) {
            this.setState({errorMessage: nextProps.errorMessage});
        }
    }
    render() {

        return (
            <CreateRosterDiv>
                <div>
                    <input type='text' onChange={this.onChangeRosterName} value={this.state.rosterName} placeholder='Enter a Roster Name' />
                </div>
                <div>
                    {this.state.errorMessage ? <ErrorMessage>{this.state.errorMessage}</ErrorMessage> : null}
                </div>
                <div>
                    <button type="button" onClick={this.onClickCreateRoster}>Create Roster</button>
                </div>
                <PlayerLegendDiv>
                    <div><strong>Player Legend:</strong></div>
                    <div>Green = Starters</div>
                    <div>Yellow = Subs</div>
                </PlayerLegendDiv>
            </CreateRosterDiv>
        )
    }
}

const mapStateToProps = state => {
    return {
        errorMessage: state.rosters.errorMessage
    }
}

export default connect(mapStateToProps)(CreateRoster);