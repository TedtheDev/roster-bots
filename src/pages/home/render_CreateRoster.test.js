import React from 'react';
import renderer from 'react-test-renderer';
import { CreateRoster } from './CreateRoster';

describe('<CreateRoster />', () => {
    it('renders correctly', () => {
        const wrapper = renderer.create(<CreateRoster />);
        const tree = JSON.stringify(wrapper);

        expect(tree).toMatchSnapshot();
    });
});