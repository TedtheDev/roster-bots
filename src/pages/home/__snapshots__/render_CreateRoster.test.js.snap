// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateRoster /> renders correctly 1`] = `"{\\"type\\":\\"div\\",\\"props\\":{\\"className\\":\\"sc-bdVaJa knkgYJ\\"},\\"children\\":[{\\"type\\":\\"div\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"input\\",\\"props\\":{\\"type\\":\\"text\\",\\"value\\":\\"\\",\\"placeholder\\":\\"Enter a Roster Name\\"},\\"children\\":null}]},{\\"type\\":\\"div\\",\\"props\\":{},\\"children\\":null},{\\"type\\":\\"div\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"button\\",\\"props\\":{\\"type\\":\\"button\\"},\\"children\\":[\\"Create Roster\\"]}]},{\\"type\\":\\"div\\",\\"props\\":{\\"className\\":\\"sc-htpNat ewtrWV\\"},\\"children\\":[{\\"type\\":\\"div\\",\\"props\\":{},\\"children\\":[{\\"type\\":\\"strong\\",\\"props\\":{},\\"children\\":[\\"Player Legend:\\"]}]},{\\"type\\":\\"div\\",\\"props\\":{},\\"children\\":[\\"Green = Starters\\"]},{\\"type\\":\\"div\\",\\"props\\":{},\\"children\\":[\\"Yellow = Subs\\"]}]}]}"`;
